File: 392.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() == 0) return true; 
        bool have = false; 
        size_t s_size = s.size(); 
        size_t marker = 0; 
        for (char c: t){
            if (c == s[marker]){
                marker++; 
                if (marker == s_size) return true; 
            }
    
        }
        return false; 
            
        
    }
};

File: 392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        # time: O(t)
        # space: O(1)
        
        s_pos = 0 
        s_size = len(s)
        if s_size == 0:
            return True 

        for index, letter in enumerate(t): 
            if letter == s[s_pos]:
                s_pos += 1
                if s_pos >= s_size: 
                    return True
                
            
        return False 
